libcudacxx/include/cuda/std/detail/libcxx/include/cmath

```diff
@@ -674,7 +674,7 @@ __constexpr_isfinite(_A1 __lcpp_x) noexcept
     return isfinite(__lcpp_x);
 }
 
-#if defined(_MSC_VER) || defined(__CUDACC_RTC__)
+#if defined(_MSC_VER) || defined(__CUDACC_RTC__) || defined(_LIBCUDACXX_COMPILER_CLANG_CUDA)
```

Comment:
@miscco I have no idea what I'm doing. Please, take a look.

Reply:
It seems that clang builtins are not available on device:
```cpp
__global__ void kernel(double *ptr) {
    ptr[0] = __builtin_logb(42.0);
}

int main() {
    kernel<<<1, 1>>>(nullptr);
}
```
results in:
```
ptxas fatal   : Unresolved extern function 'logb'
clang: error: ptxas command failed with exit code 255 (use -v to see invocation)
```

Reply:
We do not have the GPU-side standard library these builtins would normally end up calling. Some of the math calls do end up being replaced by their `__nv_*` counterparts, but there's none for `logb` and that's why you see the unresolved reference.

Reply:
Is this something we could contribute? I guess there is a special list of builtins and how they are delegated?